network/endpoint_windows.go

```diff
@@ -44,6 +57,19 @@ func ConstructEndpointID(containerID string, netNsPath string, ifName string) (s
 
 // newEndpointImpl creates a new endpoint in the network.
 func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
+	if useHnsV2, err := UseHnsV2(epInfo.NetNsPath); useHnsV2 {
+		if err != nil {
+			return nil, err
+		}
+
+		return nw.newEndpointImplHnsV2(epInfo)
+	}
+
+	return nw.newEndpointImplHnsV1(epInfo)
```

Comment:
we can call ConstructEndpointID straight from struct init if infraEpName isn't used elsewhere